VirtoCommerce.PricingModule.Data/Services/PricingServiceImpl.cs

```diff
@@ -425,6 +427,27 @@ public virtual void DeletePricelistsAssignments(string[] ids)
                 ResetCache();
             }
         }
+
+        public virtual void DeletePricelistsAssignmentsByFilter(PricelistAssignmentsSearchCriteria criteria)
```

Comment:
As @tatarincev mentioned, we do not need new service method at all, we could use existing search and delete with paging in controller in DeleteFilteredAssignments.

Reply:
Made it to use remove by ids method